lib/api/unittest/CDataFrameAnalysisRunnerTest.cc

```diff
@@ -91,9 +93,10 @@ void CDataFrameAnalysisRunnerTest::testComputeAndSaveExecutionStrategyDiskUsageF
         api::CDataFrameAnalysisSpecification spec{jsonSpec};
 
         // single error is registered that the memory limit is to low
+        LOG_DEBUG(<< "errors = " << core::CContainerPrinter::print(errors));
+        std::regex re{"Input error: memory limit.*"};
```

Comment:
I don't think we should use `std::regex`.  There are numerous reports of it being slow compared to Boost, for example:

* https://stackoverflow.com/questions/51503590/stdregex-is-much-slower-than-boostregex-on-vc2015u3
* https://stackoverflow.com/questions/20942450/why-c11-regex-libc-implementation-is-so-slow
* https://stackoverflow.com/questions/14205096/c11-regex-slower-than-python

I think we should stick with the `core::CRegex` class.

Reply:
This isn't a performance bottleneck: it is test only and instantaneous. I do wonder if performance of regex would really be important for any of our code. If it is there are better libraries than boost from that perspective, e.g. hscan, if it isn't isn't it better to cut down our dependencies. However, I guess the argument would be that we should be consistent with the rest of the code and that probably uses `core::CRegex`, so I'm happy to switch this.

Reply:
> if it isn't isn't it better to cut down our dependencies

This change doesn't cut the dependencies, because it hasn't removed every single usage of `boost::regex` from the code.

> there are better libraries than boost from that perspective, e.g. hscan

That's a good argument for always using `core::CRegex` because then if we switch over then we just change it in one place.

> This isn't a performance bottleneck

True, but it sets the precedent that it's OK to use `std::regex` directly.

Reply:
[ad16672](https://github.com/elastic/ml-cpp/pull/553/commits/ad16672db1881e47791f94a655386fc24ccf25d1)